src/activities/Elsa.Activities.Http/Middleware/HttpEndpointMiddleware.cs

```diff
@@ -75,7 +75,11 @@ public class HttpEndpointMiddleware
                 return;
             }
 
-            var workflowBlueprint = await workflowRegistry.FindAsync(x => x.IsPublished && x.Id == pendingWorkflowInstance.DefinitionId && !x.IsDisabled, cancellationToken);
+            var isTest = pendingWorkflowInstance.MetaData.FirstOrDefault(x => x.Key == "isTest").Value;
+            var workflowBlueprint = (isTest != null && Convert.ToBoolean(isTest)) ?
+                await workflowRegistry.FindAsync(x => x.Id == pendingWorkflowInstance.DefinitionId, cancellationToken) :
+                await workflowRegistry.FindAsync(x => x.IsPublished && x.Id == pendingWorkflowInstance.DefinitionId && !x.IsDisabled, cancellationToken);
```

Comment:
Similar to my other comment, this code will make testing work for HTTP endpoint activities, but not for all of the other trigger activities - they too try to get a workflow blueprint, but right now they are not aware of any "isTest" metadata. They will fail to find any unpublished workflow blueprints.

A neat solution to this problem would be to implement a decorator around the `IWorkflowRegistry` service that exists in the `Elsa.Server.Api` project as well (since this should be the only library containing test-related code).

The decorator should override a new method to be created on `IWorkflowRegistry` called `FindActiveAsync` (representing the "singular" sibling of `ListActiveAsync` that already exists and returns a list of results).

The overridden method would then simply replicate what is done in the decorated service, but include the "isTest" logic to include blueprints under test even if they are not published (like you're doing now).

Reply:
I think we may not need a new decorator. If we have this code in WorkflowRegistry:
`
        public async Task<IWorkflowBlueprint?> FindActiveAsync(Func<IWorkflowBlueprint, bool> predicate, CancellationToken cancellationToken) =>
            (await ListActiveAsync(cancellationToken).Where(predicate).OrderByDescending(x => x.Version)).FirstOrDefault();

        private async IAsyncEnumerable<IWorkflowBlueprint> ListActiveInternalAsync([EnumeratorCancellation] CancellationToken cancellationToken)
        {
            var workflows = await ListAsync(cancellationToken);

            foreach (var workflow in workflows)
            {
                // If a workflow is not published, only consider it for processing if it has at least one non-ended workflow instance.
                if (!workflow.IsPublished && !await WorkflowHasUnfinishedWorkflowsAsync(workflow, cancellationToken))
                    continue;

                yield return workflow;
            }
        }
`
it works fine for IsTest scenario too if we change the code in HttpEndpointMiddleware.cs to that one:
`
var workflowBlueprint = await workflowRegistry.FindActiveAsync(x => x.Id == pendingWorkflowInstance.DefinitionId && !x.IsDisabled, cancellationToken);
`

Reply:
But isn't the problem with that change, that now even un-published workflows will be invoked?

Reply:
As just discussed over skype, the concern I raised previously is not actually a problem. As it turns out, the HTTP Middleware already requires a custom bit of logic to load the workflow blueprint. For other trigger sources such as Timer, they simply schedule work for the corresponding workflow instance, which in turn get a specific workflow blueprint by version.

Sorry for the confusion.